### ISSUE TITLE
crash

### ISSUE THREAD

**Issue by alfianguide** [flutter/flutter#26834]
Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command

flutter 

## exception

FileSystemException: FileSystemException: Cannot open file, path = '/opt/flutter/bin/cache/flutter_version_check.stamp' (OS Error: Permission denied, errno = 13)

```
#0      _File.throwIfError (dart:io/file_impl.dart:647:7)
#1      _File.openSync (dart:io/file_impl.dart:491:5)
#2      _File.writeAsBytesSync (dart:io/file_impl.dart:616:31)
#3      _File.writeAsStringSync (dart:io/file_impl.dart:640:5)
#4      _LocalFile&LocalFileSystemEntity&ForwardingFile.writeAsStringSync (package:file/src/forwarding/forwarding_file.dart:147:16)
#5      Cache.setStampFor (package:flutter_tools/src/cache.dart:176:35)
#6      VersionCheckStamp.store (package:flutter_tools/src/version.dart:453:20)
<asynchronous suspension>
#7      FlutterVersion._getLatestAvailableFlutterDate (package:flutter_tools/src/version.dart:375:31)
<asynchronous suspension>
#8      FlutterVersion.checkFlutterVersionFreshness (package:flutter_tools/src/version.dart:280:52)
<asynchronous suspension>
#9      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:377:41)
<asynchronous suspension>
#10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:29)
<asynchronous suspension>
#11     _rootRun (dart:async/zone.dart:1124:13)
#12     _CustomZone.run (dart:async/zone.dart:1021:19)
#13     _runZoned (dart:async/zone.dart:1516:10)
#14     runZoned (dart:async/zone.dart:1463:12)
#15     AppContext.run (package:flutter_tools/src/base/context.dart:141:18)
<asynchronous suspension>
#16     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:361:19)
<asynchronous suspension>
#17     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:111:29)
#18     new Future.sync (dart:async/future.dart:224:31)
#19     CommandRunner.run (package:args/command_runner.dart:111:11)
#20     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:241:18)
#21     run.<anonymous closure> (package:flutter_tools/runner.dart:60:20)
<asynchronous suspension>
#22     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:29)
<asynchronous suspension>
#23     _rootRun (dart:async/zone.dart:1124:13)
#24     _CustomZone.run (dart:async/zone.dart:1021:19)
#25     _runZoned (dart:async/zone.dart:1516:10)
#26     runZoned (dart:async/zone.dart:1463:12)
#27     AppContext.run (package:flutter_tools/src/base/context.dart:141:18)
<asynchronous suspension>
#28     runInContext (package:flutter_tools/src/context_runner.dart:43:24)
<asynchronous suspension>
#29     run (package:flutter_tools/runner.dart:51:10)
#30     main (package:flutter_tools/executable.dart:51:9)
<asynchronous suspension>
#31     main (file:///b/build/slave/Linux_Flutter_Packaging/build/archive/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#32     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:287:32)
#33     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

## flutter doctor

```
[32m[✓][39m Flutter (Channel stable, v1.0.0, on Linux, locale en_US.UTF-8)
    [32m•[39m Flutter version 1.0.0 at /opt/flutter
    [32m•[39m Framework revision 5391447fae (7 weeks ago), 2018-11-29 19:41:26 -0800
    [32m•[39m Engine revision 7375a0f414
    [32m•[39m Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[33m[!][39m Android toolchain - develop for Android devices (Android SDK 28.0.3)
    [32m•[39m Android SDK at /home/alfianguide/Android/Sdk
    [32m•[39m Android NDK location not configured (optional; useful for native profiling
      support)
    [32m•[39m Platform android-28, build-tools 28.0.3
    [32m•[39m Java binary at: /opt/android-studio/jre/bin/java
    [32m•[39m Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1248-b01)
    [31m✗[39m Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses

[32m[✓][39m Android Studio (version 3.3)
    [32m•[39m Android Studio at /opt/android-studio
    [32m•[39m Flutter plugin version 31.3.3
    [32m•[39m Dart plugin version 182.5124
    [32m•[39m Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1248-b01)

[33m[!][39m Connected device
    [33m![39m No devices available

[33m![39m Doctor found issues in 2 categories.
```

**Comment by alfianguide** [comment]
Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command

flutter 

## exception

FileSystemException: FileSystemException: Cannot open file, path = '/opt/flutter/bin/cache/flutter_version_check.stamp' (OS Error: Permission denied, errno = 13)

```
#0      _File.throwIfError (dart:io/file_impl.dart:647:7)
#1      _File.openSync (dart:io/file_impl.dart:491:5)
#2      _File.writeAsBytesSync (dart:io/file_impl.dart:616:31)
#3      _File.writeAsStringSync (dart:io/file_impl.dart:640:5)
#4      _LocalFile&LocalFileSystemEntity&ForwardingFile.writeAsStringSync (package:file/src/forwarding/forwarding_file.dart:147:16)
#5      Cache.setStampFor (package:flutter_tools/src/cache.dart:176:35)
#6      VersionCheckStamp.store (package:flutter_tools/src/version.dart:453:20)
<asynchronous suspension>
#7      FlutterVersion._getLatestAvailableFlutterDate (package:flutter_tools/src/version.dart:375:31)
<asynchronous suspension>
#8      FlutterVersion.checkFlutterVersionFreshness (package:flutter_tools/src/version.dart:280:52)
<asynchronous suspension>
#9      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:377:41)
<asynchronous suspension>
#10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:29)
<asynchronous suspension>
#11     _rootRun (dart:async/zone.dart:1124:13)
#12     _CustomZone.run (dart:async/zone.dart:1021:19)
#13     _runZoned (dart:async/zone.dart:1516:10)
#14     runZoned (dart:async/zone.dart:1463:12)
#15     AppContext.run (package:flutter_tools/src/base/context.dart:141:18)
<asynchronous suspension>
#16     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:361:19)
<asynchronous suspension>
#17     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:111:29)
#18     new Future.sync (dart:async/future.dart:224:31)
#19     CommandRunner.run (package:args/command_runner.dart:111:11)
#20     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:241:18)
#21     run.<anonymous closure> (package:flutter_tools/runner.dart:60:20)
<asynchronous suspension>
#22     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:29)
<asynchronous suspension>
#23     _rootRun (dart:async/zone.dart:1124:13)
#24     _CustomZone.run (dart:async/zone.dart:1021:19)
#25     _runZoned (dart:async/zone.dart:1516:10)
#26     runZoned (dart:async/zone.dart:1463:12)
#27     AppContext.run (package:flutter_tools/src/base/context.dart:141:18)
<asynchronous suspension>
#28     runInContext (package:flutter_tools/src/context_runner.dart:43:24)
<asynchronous suspension>
#29     run (package:flutter_tools/runner.dart:51:10)
#30     main (package:flutter_tools/executable.dart:51:9)
<asynchronous suspension>
#31     main (file:///b/build/slave/Linux_Flutter_Packaging/build/archive/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#32     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:287:32)
#33     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

## flutter doctor

```
[32m[✓][39m Flutter (Channel stable, v1.0.0, on Linux, locale en_US.UTF-8)
    [32m•[39m Flutter version 1.0.0 at /opt/flutter
    [32m•[39m Framework revision 5391447fae (7 weeks ago), 2018-11-29 19:41:26 -0800
    [32m•[39m Engine revision 7375a0f414
    [32m•[39m Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[32m[✓][39m Android toolchain - develop for Android devices (Android SDK 28.0.3)
    [32m•[39m Android SDK at /home/alfianguide/Android/Sdk
    [32m•[39m Android NDK location not configured (optional; useful for native profiling
      support)
    [32m•[39m Platform android-28, build-tools 28.0.3
    [32m•[39m Java binary at: /opt/android-studio/jre/bin/java
    [32m•[39m Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1248-b01)
    [32m•[39m All Android licenses accepted.

[32m[✓][39m Android Studio (version 3.3)
    [32m•[39m Android Studio at /opt/android-studio
    [32m•[39m Flutter plugin version 31.3.3
    [32m•[39m Dart plugin version 182.5124
    [32m•[39m Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1248-b01)

[33m[!][39m Connected device
    [33m![39m No devices available

[33m![39m Doctor found issues in 1 category.
```

**Comment by alfianguide** [comment]
it cause new error

**Comment by zoechi** [comment]
Flutter currently doesn't support being installed in a directory that requires elevated privileges for write access.

**Comment by zoechi** [comment]
I'm going to close because this looks like a support request.
Feel free to continue the discussion though.

**Comment by alfianguide** [comment]
it fixed, thx.

**Comment by Poohblah** [comment]
@kangwang1988 may I ask that you refrain from suggesting `chmod -R 777`? [It is bad practice, a potential security concern, and generally displays a lack of understanding of unix filesystem permissions](https://askubuntu.com/questions/20105/why-shouldnt-var-www-have-chmod-777/20110#20110). :)

**Comment by zoechi** [comment]
@Poohblah I'm all but an expert on Linux file permissions but the issues mentioned in the linked ServerFault page seem all to be related to applying `chmod -R 777` to `/` and don't seem to apply if it's done to a specific directory.

**Comment by Poohblah** [comment]
@zoechi I don't want to derail this thread too much so I'll explain in my own words and hopefully we can move on.

I'm a Linux sysadmin and hearing people suggest `chmod 777` is one of my biggest pet peeves.  Occasionally we give out admin privileges to our users, and seeing a user run `chmod 777` is grounds for immediately revoking their admin privileges, as not only has the user potentially opened up security holes, but also it is one of the most obvious signs that a user does not understand what they are doing nor the implications of altering filesystem permissions.

In my personal opinion, there are three major issues with `chmod 777`: 

1. It's best security practice to only grant exactly as much access as a user or process requires.  For instance, if you're a landlord and you need to hand over keys to a new tenant, you wouldn't hand them over your entire keychain, even if the tenant promises that they'll be really really nice and never access their neighbors' apartments or do any other bad thing.  Giving somebody those keys that they don't need is a huge liability concern.   `chmod 777` is like handing copies of all of your keys to all of your tenants.

    Giving read, write, and execute permissions to all users is obviously a security issue on a multi-user system.  It's less obvious that this is a security issue on a single-purpose server or on a personal system such as a laptop or desktop.  Even on personal systems, different processes and daemons run as different users and groups, which helps to isolate and sandbox processes.  Just to choose a completely arbitrary example, on my desktop the system logging daemon runs as its own user.  If I run `chmod 777` on files in my software project directory, then the logging daemon can now read and write those files.  You might argue that the logging daemon, when properly configured, won't even attempt to write to files that it doesn't need to, but that's not really the point.  If there's a misconfiguration in the logging daemon, or a software bug, or a security vulnerability that allows remote access, then the logging daemon can destroy files in my software project or read files (like private keys) that it should have no right to read.

2.  `chmod 777` is often suggested by people who use it as a quick catch-all solution to filesystem errors without taking the time to actually understand the error and come up with a proper solution.  If a single user needs to read and write files in a directory, the appropriate solution is to use `chown -R` to give ownership to a single user.  If a small number of users need read/write access, create a group for those users (if one doesn't already exist), and use `chgrp` and `chmod g+rwX` to only give access to members of that group.  It's extremely rare that every single user on a system - including system users - will need read/write access to a directory recursively, but if they did, the appropriate solution would be `chmod -R +rwX`, which (among other differences with `777`) only grants execute permissions on files and directories that need it.

    `chmod 777` grants execute permissions in addition to read/write.  Execute permissions are necessary only on directories and on executable files.  They are completely unnecessary on files that are not intended to be executed, which is pretty much everything outside of a `bin` directory on a unix system.  Giving execute permissions to a non-executable file is generally not a technical problem, but it does strongly suggest that the user who applied execute permissions to the file doesn't understand when it is appropriate or inappropriate to have the executable permission set.

    It's extraordinarily rare that `777` is the appropriate permission set to apply to a file or directory.  In my sysadmin experience, the closest I have seen is `1777`.  Usually `1777` is applied to `/tmp` (just the top-level directory, not recursively).  `/tmp` is shared among all processes on a system, with the caveat that the sticky bit (the leading `1`) prevents processes from deleting files/directories that they do not own.

3. `777` is destructive.  It destroys sticky, setuid, and setgid bits.  It can also destroy ACLs.  These are often necessary for the proper behavior of a file or directory.  `chmod +rwx` does the same thing as `777` without destroying other filesystem permissions - so why ever use `777`?

In a nutshell, though, I would say that suggesting `chmod 777` is an immediate giveaway that the person who made the suggestion has no idea what they're doing.  Perhaps the biggest issue that I have with `chmod 777` is that this command tends to get passed down from authority figures (like popular bloggers or developers on popular software projects) to new users, who then use and share it far too liberally without fully understanding what it does, thinking that because it was given to them by the authority figure and it solved their issue that it must be a good thing worth sharing.

I suppose it's also worth noting that this applies generically to all unix systems (including Mac OS), not just Linux.

/rant :)

**Comment by alfianguide** [comment]
Sorry guys, i'm realized now, i was stupid that time and don't know how to deal with it. But it's not a big deal 🙏 thx for responding btw. But please, the case closed 🙂🙏

**Comment by btd1337** [comment]
For those who are experiencing this problem, run the following command and try again.

```
sudo chown -R $USER /opt/flutter/
```

Change `/opt/flutter` to your flutter instalation folder.

**Comment by tomay3000** [comment]
> @alfianguide
> Delete `/opt/flutter/bin/cache/flutter_version_check.stamp` and try again.

This solved it for me under `Manjaro Linux v19.2`

Thanks ;)
